libraries/core_libs/consensus/src/dag/dag_block_manager.cpp

```diff
@@ -206,97 +165,59 @@ void DagBlockManager::pushVerifiedBlock(DagBlock const &blk) {
-      while (unverified_qu_.empty() && !stopped_) {
-        cond_for_unverified_qu_.wait(lock);
-      }
-      if (stopped_) {
-        return;
-      }
-      blk = unverified_qu_.begin()->second.front();
-      unverified_qu_.begin()->second.pop_front();
-      if (unverified_qu_.begin()->second.empty()) {
-        unverified_qu_.erase(unverified_qu_.begin());
-      }
-    }
-
-    const auto &block_hash = blk.getHash();
-    LOG(log_time_) << "Verifying Trx block  " << block_hash << " at: " << getCurrentTimeMilliSeconds();
-
-    if (invalid_blocks_.count(block_hash)) {
-      LOG(log_dg_) << "Skip invalid DAG block " << block_hash;
-      continue;
-    }
+bool DagBlockManager::verifyBlock(const DagBlock &blk) {
+  const auto &block_hash = blk.getHash();
+  LOG(log_time_) << "Verifying Trx block  " << block_hash << " at: " << getCurrentTimeMilliSeconds();
 
-    // Verify transactions
-    if (!trx_mgr_->checkBlockTransactions(blk)) {
-      LOG(log_er_) << "Ignore block " << block_hash << " since it has missing transactions";
-      // This can be a valid block so just remove it from the seen list
-      seen_blocks_.erase(block_hash);
-      continue;
-    }
+  if (invalid_blocks_.count(block_hash)) {
+    LOG(log_wr_) << "Skip invalid DAG block " << block_hash;
+    return false;
+  }
 
-    auto propose_period = getProposalPeriod(blk.getLevel());
-    // Verify DPOS
-    if (!propose_period.second) {
-      // Cannot find the proposal period in DB yet. The slow node gets an ahead block, remove from seen_blocks
-      LOG(log_nf_) << "Cannot find proposal period " << propose_period.first << " in DB for DAG block "
-                   << blk.getHash();
-      seen_blocks_.erase(block_hash);
-      continue;
-    }
+  // Verify transactions
+  if (!trx_mgr_->checkBlockTransactions(blk)) {
+    LOG(log_er_) << "Ignore block " << block_hash << " since it has missing transactions";
+    // This can be a valid block so just remove it from the seen list
+    seen_blocks_.erase(block_hash);
+    return false;
+  }
 
-    // Verify VDF solution
-    try {
-      blk.verifyVdf(sortition_params_manager_.getSortitionParams(propose_period.first));
-    } catch (vdf_sortition::VdfSortition::InvalidVdfSortition const &e) {
-      LOG(log_er_) << "DAG block " << block_hash << " with " << blk.getLevel()
-                   << " level failed on VDF verification with pivot hash " << blk.getPivot() << " reason " << e.what();
-      LOG(log_er_) << "period from map: " << propose_period.first << " current: " << pbft_chain_->getPbftChainSize();
-      markBlockInvalid(block_hash);
-      continue;
-    }
+  auto propose_period = getProposalPeriod(blk.getLevel());
+  // Verify DPOS
+  if (!propose_period.second) {
+    // Cannot find the proposal period in DB yet. The slow node gets an ahead block, remove from seen_blocks
+    LOG(log_er_) << "Cannot find proposal period " << propose_period.first << " in DB for DAG block " << blk.getHash();
```

Comment:
I would not use error log here, this is something that can happen sometimes and you could have a lot of these logs all at once when node just joins the network and it is way behind but has not started syncing yet and then community might complain about it

Reply:
so warn level?

Reply:
I would go with info, I see it was info before your change but warn is fine as well